kroxylicious/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -53,6 +61,19 @@ String filterDescriptor() {
 
     @Override
     public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
+        if (msg instanceof InternalRequestFrame<?>) {
+            // jump the queue, internal request must flow!
+            doWrite(ctx, msg, promise);
+        }
+        else if (writeFuture.isDone()) {
+            writeFuture = doWrite(ctx, msg, promise);
```

Comment:
to self - I suppose we are doing this to avoid a big long chain of completed futures.  it should be safe, but makes me worry we might lose track of something.